src/jit/morph.cpp

```diff
@@ -4452,10 +4452,8 @@ GenTreeCall* Compiler::fgMorphArgs(GenTreeCall* call)
 #endif // FEATURE_FIXED_OUT_ARGS
 
     /* Update the 'side effect' flags value for the call */
-
-    call->gtFlags |= (flagsSummary & GTF_ALL_EFFECT);
-
-    if (!call->OperMayThrow(this) && ((flagsSummary & GTF_EXCEPT) == 0))
+    call->gtFlags = (call->gtFlags & ~GTF_ASG) | (flagsSummary & GTF_ALL_EFFECT);
+    if (((flagsSummary & GTF_EXCEPT) == 0) && !call->OperMayThrow(this))
```

Comment:
I believe that the GTF_EXCEPT flag can be handled similarly to the GTF_ASG flag:

And this would read better if we have two operations:
1) clearing flags from the call node
2) setting flags from its child node

```
 // Clear the GTF_ASG and possibly the GTF_EXCEPT flags on the call node
 call->gtFlags &=  ~GTF_ASG;
if (!call->OperMayThrow(this))
{
    call->gtFlags &=  ~GTF_EXCEPT;
}

 // Update the 'side effect' flags value for the call node from its child nodes
 call->gtFlags |= (flagsSummary & GTF_ALL_EFFECT);

```

Reply:
I like Brian's version.

Reply:
That works for me; I'll make the change